api/v2/Routes/students/grades/index.js

```diff
@@ -6,16 +6,33 @@ import { isAdmin } from '../../../../lib/userlib.mjs';
 const router = Router({ mergeParams: true });
 
 router.get('/', async (req, res) => {
-    const { id } = req.params;
-    const maxScores = await getMaxScores();
-    let studentScores;
-    if (isAdmin(id)) {
-        studentScores = maxScores;
-    } else {
-        studentScores = await getStudentScores(id);
+    const { id } = req.params; // the id is the student's email
+    let maxScores;
+    try {
+        // Attempt to get max scores, if fails, return 404
+        maxScores = await getMaxScores();
+    } catch (error) {
+        return res.status(404).json({ message: 'Error fetching max scores' });
```

Comment:
This I think a bit too much of a blanket -- if getMaxScores errors for another reason, it should be a diff error (eg: if the db is down, it should throw a 500)